traits/trait_set_object.py

```diff
@@ -355,16 +333,15 @@ def intersection_update(self, value):
 
 
         """
-        validated_values = self.validate(value)
-        removed = self.difference(self.intersection(validated_values))
+        if value is None:
+            return
+        removed = self.difference(value)
         if len(removed) > 0:
-            super().intersection_update(validated_values)
+            super().intersection_update(value)
             self.notify(removed, set())
 
     def symmetric_difference_update(self, value):
-        """ Return the symmetric difference of two sets as a new set.
-
-        (i.e. all elements that are in exactly one of the sets.)
+        """ Update a set with the symmetric difference of itself and another.
```

Comment:
This docstring is copied from Python set, likewise for the one for `intersection_update`.